codemods/antd/5/utils/marker.js

```diff
@@ -0,0 +1,68 @@
+/*! @license
+MIT LICENSE
+
+Copyright (c) 2015-present Ant UED, https://xtech.antfin.com/
+
+Permission is hereby granted, free of charge, to any person obtaining
+a copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the following conditions:
+
+The above copyright notice and this permission notice shall be
+included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+The source code has been taken from https://github.com/ant-design/codemod-v5/blob/main/transforms/utils/marker.js
+
+Changes to the original file: changed imports to esm 
+*/
+
+import fs from 'fs';
+import path from 'path';
+import uniq from 'lodash/uniq';
+
+const markerPath = path.join(
+  process.env.NODE_ENV === 'local' ? process.cwd() : require('os').tmpdir(),
+  './antd5-codemod-marker.log',
+);
+
+const newline = '\n';
+
+function ensureFile() {
+  return fs.openSync(markerPath, 'w');
+}
+
+async function cleanup() {
+  return await fs.promises.unlink(markerPath);
+}
+
+function markDependency(depName) {
+  ensureFile();
+  return fs.appendFileSync(
+    markerPath,
+    depName + newline,
+    'utf8'
+  );
+}
+
+async function getDependencies() {
```

Comment:
is this used anywhere? I think it will crash as direct FS operations are unsupported in our model of execution.

Reply:
only markDependency is used, but its also using `fs`. Should we reimplement this codemod as Filemod?

Reply:
I would remove the parts that are not used and comment out the usage of mark dependency. We may consider moving to a filemod later on.